### ISSUE TITLE
why keep discriminator_model.trainable=False all the time?

### ISSUE THREAD

**Issue by USTClj** [fchollet/deep-learning-with-python-notebooks#13]
I have the same issue as:  https://github.com/wayaai/SimGAN/issues/6

    refiner_model.compile(optimizer=sgd, loss=self_regularization_loss)
    discriminator_model.compile(optimizer=sgd, loss=local_adversarial_loss)
    discriminator_model.trainable = False
    combined_model.compile(optimizer=sgd, loss=[self_regularization_loss, local_adversarial_loss])

I think when training the discriminator network, It should be change to True and False alternately.
And I haven't gone deeper in the mechanism of keras, here if my understanding and I'm not sure if I'm wrong.

I guess that when you specify the trainable = True after compiling the discriminator, the trainable state will not be changed even if you use discriminator_model.trainable = False.
The combined model consists of trainable refiner model and the untrainable discriminate model. So we needn't change the trainable state alternatively during training which is different to many other example such as : https://github.com/osh/KerasGAN/blob/master/MNIST_CNN_GAN_v2.ipynb
